mira/dkg/web_client.py

```diff
@@ -1,104 +1,214 @@
+    res_json = web_client(
+        endpoint="/relations", method="post", query_json=query_json, api_url=api_url
+    )
+    if res_json is not None:
+        if relations_model.full:
+            return [api.FullRelationResponse(**r) for r in res_json]
+        else:
+            return [api.RelationResponse(**r) for r in res_json]
+
+
+def get_entity_web(curie: str, api_url: Optional[str] = None) -> Optional[api.Entity]:
+    """Get information about an entity based on its compact URI (CURIE)
+
+    A wrapper that calls the REST API's entity endpoint.
+
+    Parameters
+    ----------
+    curie :
+        The curie for an entity to get information about.
+    api_url :
+        Use this parameter to specify the REST API base url or to override
+        the url set in the environment or the config.
+
+
+    Returns
+    -------
+    :
+        Returns an Entity model, if the entity exists in the graph.
+    """
+    res_json = web_client(endpoint=f"/entity/{curie}", method="get", api_url=api_url)
+    if res_json is not None:
+        return api.Entity(**res_json)
+
+
+def get_lexical_web(api_url: Optional[str] = None) -> List[Dict[str, Any]]:
+    """Get lexical information for all entities in the graph
+
+    A wrapper that calls the REST API's lexical endpoint.
 
     Parameters
     ----------
-    source_type :
-        The source type (i.e., prefix). Example: "vo".
-    source_curie :
-        The source compact URI (CURIE). example="doid:946".
-    target_type :
-        "The target type (i.e., prefix)", example="ncbitaxon"
-    target_curie :
-        "The target compact URI (CURIE)", example="ncbitaxon:10090"
-    relations :
-        "A relation string or list of relation strings", example="vo:0001243"
-    relation_direction :
-        "The direction of the relationship". Options are "left", "right" and "both". Default: "right".
-    relation_min_hops :
-        "The minimum number of relationships between the subject and
-        object.". Default: 1.
-    relation_max_hops :
-        The maximum number of relationships between the subject and object.
-        Set to 0 to make infinite. Default: 1
-    limit :
-        A limit on the number of records returned. Example=50. Default: no
-        limit.
-    full :
-        A flag to turn on full entity and relation metadata return. Warning, this gets pretty verbose. Default: False.
-    distinct :
-        A flag to turn on the DISTINCT flag in the return of a cypher query. Default: False
     api_url :
         Use this parameter to specify the REST API base url or to override
-        the url set
+        the url set in the environment or the config.
 
     Returns
     -------
+    :
+        A list of all entities in the graph.
+    """
+    return web_client(endpoint="/lexical", method="get", api_url=api_url)
+
+
+def ground_web(
+    grounding_model: grounding.GroundRequest, api_url: Optional[str] = None
```

Comment:
This definition is a little bit odd, I think we should just allow the basic grounding parameters like text, context, namespaces directly as parameters.

Reply:
I reverted this in https://github.com/indralab/mira/pull/38/commits/35b339c39a4f3c899711aa495f1d4381623a244e